client/gandi.go

```diff
@@ -11,33 +11,49 @@ import (
 )
 
 const (
-	gandiEndpoint = "https://dns.api.gandi.net/api/v5/"
-	// HTTP Methods
-	mPATCH  = http.MethodPatch
-	mGET    = http.MethodGet
-	mPOST   = http.MethodPost
-	mDELETE = http.MethodDelete
-	mPUT    = http.MethodPut
+	gandiEndpoint = "https://api.gandi.net/v5/"
 )
 
-// Gandi makes it easier to interact with Gandi LiveDNS
+// Gandi makes it easier to interact with the Gandi API
 type Gandi struct {
 	apikey     string
 	sharing_id string
+	endpoint string
 	debug      bool
+	dryRun bool
 }
 
 // New instantiates a new Gandi instance
-func New(apikey string, sharing_id string) *Gandi {
-	return &Gandi{apikey: apikey, sharing_id: sharing_id}
+func New(apikey string, sharing_id string, debug bool, dry_run bool) *Gandi {
```

Comment:
Every time you add a new feature the signature of this function will change.  How about passing the apikey plus a struct:

``` go
type Options {
      sharing_id string
	endpoint string
	debug      bool
	dryRun bool
}

gclient := gandi.New('my-secret-key", gandi.Options{ dryRun: true })

```

Reply:
( or a better way to pass options :) )